lib/micro/case.rb

```diff
@@ -239,6 +239,21 @@ def Failure(type = :error, result: nil)
         __get_result(false, value, type)
       end
 
+
+      def Check(type = nil, result: nil, on: {})
```

Comment:
@MatheusRich you could use a constant with a frozen empty hash to avoid unnecessary object allocations (defaults always create new instances). Follows a suggestion using `Kind::Empty::HASH` to address this kind of issue.

```suggestion
      def Check(type = nil, result: nil, on: Kind::Empty::HASH)
```